docs/designs/login-design.md

```diff
@@ -0,0 +1,165 @@
+# Architecture Design Document
+
+## 1. Introduction
+
+### 1.1 Purpose
+This document outlines the design of a sign-in page for PH admins and volunteers. Users will be authenticated through Azure OAuth and redirected based on their role (admin or volunteer). The document focuses on the design of the sign-in page, the volunteer PIN page, and the supporting database schema.
+
+### 1.2 Scope
+The scope of this design includes the development of:
+- A sign-in page redirecting to Azure OAuth for authentication.
+- Role-based redirection:  admins are directed to the dashboard, while volunteers are directed to a name selection page before entering their 4-digit PIN.
+- A volunteer name page where volunteers select their name and enter a 4-digit PIN for sign-in.
+- A mechanism to retrieve forgotten PINs via email.
+- A database schema supporting volunteer sign-ins.
+
+The functionalities of the admin dashboard, volunteer dashboard, setting user activation status, and enrolling or deleting users are outside the scope of this document.
+
+### 1.3 Definitions, Acronyms, and Abbreviations
+- **OAuth**: Open Authorization, a protocol for secure user authentication.
+- **PIN**: Personal Identification Number.
+
+### 1.4 References
+- Azure OAuth Documentation
+- User Interface Design Guidelines
+
+## 2. System Overview
+The system authenticates users via Azure OAuth, enforcing that the email domain is @plymouthhousing.org. After authentication:
+- Volunteers are redirected to a PIN page (such as volunteers@plymouthhousing.org).
+- Administrators are redirected to the admin dashboard (for accounts not using the volunteers@plymouthhousing.org domain).
+
+Volunteers will select their name from a dropdown and enter a 4-digit PIN to sign in.
+
+## 3. Architecture Goals and Constraints
+
+### 3.1 Goals
+- Secure sign-in via Azure OAuth.
+- Role-based redirection.
+- Simple PIN-based authentication for volunteers.
+- Data integrity and security for volunteer information.
+- Mechanism for PIN recovery via email.
+
+
+### 3.2 Constraints
+- Must use Azure OAuth for authentication.
+- Admin and volunteer dashboards are outside the scope of this design.
+- The solution must comply with data protection and privacy regulations.
+- The process for resetting a volunteer's PIN has not been finalized. It could be managed by admins via the admin page or by the volunteers themselves through a new PIN reset page.
```

Comment:
Can we start with a "forgot PIN email" that just sends the PIN back to the volunteer?

Reply:
I will update the Forgot PIN flow to what we discussed today by displaying a dialog box, "Please call the PH administrator for your PIN."